src/parser/mod.rs

```diff
@@ -11173,12 +11173,36 @@ impl<'a> Parser<'a> {
                     self.peek_token().location
                 )
             }
-            expr => self
-                .parse_optional_alias(keywords::RESERVED_FOR_COLUMN_ALIAS)
-                .map(|alias| match alias {
-                    Some(alias) => SelectItem::ExprWithAlias { expr, alias },
-                    None => SelectItem::UnnamedExpr(expr),
-                }),
+            expr => {
+                // Parse a [`SelectItem`] based on an [MsSql] syntax that uses the equal sign
+                // to denote an alias, for example: SELECT col_alias = col FROM tbl
+                // [MsSql]: https://learn.microsoft.com/en-us/sql/t-sql/queries/select-examples-transact-sql?view=sql-server-ver16#b-use-select-with-column-headings-and-calculations
+                let expr = if self.dialect.supports_eq_alias_assigment() {
+                    if let Expr::BinaryOp {
+                        ref left,
+                        op: BinaryOperator::Eq,
+                        ref right,
+                    } = expr
+                    {
+                        if let Expr::Identifier(alias) = left.as_ref() {
+                            return Ok(SelectItem::ExprWithAlias {
+                                expr: *right.clone(),
+                                alias: alias.clone(),
+                            });
+                        }
+                    }
+                    expr
+                } else {
+                    expr
+                };
+
+                // Parse the common AS keyword for aliasing a column
+                self.parse_optional_alias(keywords::RESERVED_FOR_COLUMN_ALIAS)
+                    .map(|alias| match alias {
+                        Some(alias) => SelectItem::ExprWithAlias { expr, alias },
+                        None => SelectItem::UnnamedExpr(expr),
+                    })
```

Comment:
```suggestion
let (expr, alias) = match expr {
    Expr::BinaryOp {
        left: Expr::Identifier(alias),
        BinaryOperator::Eq,
        right,
    } if self.dialect.supports_eq_alias_assigment() => {
        (expr, Some(alias))
    }
    expr => {
        // Parse the common AS keyword for aliasing a column
        self.parse_optional_alias(keywords::RESERVED_FOR_COLUMN_ALIAS)?
    }
};
Ok(alias
    .map(|alias| SelectItem::ExprWithAlias{ expr, alias })
    .unwrap_or_else(|| SelectItem::UnnamedExpr(expr)))
```
The current inline looks reasonable, I think its mostly rustfmt that making it look a like it has bit more code than it does, but I think we could simplify the logic a bit as above to help with that and readability/intent as well? hopefully?

Reply:
I don't think you can match to `Expr::Identifier` in L11182 because `left` is a boxed field. This further complicates the logic of the function because you now need another `if let` condition to check the value of `left`. The complexity ends up similar to what I suggested.

I suggest going back to the helper function design, it cleanly encapsulates this rather edgy case of MsSql without complicating the main flow of parsing the common syntax for aliasing a column.

Reply:
this works as its own match clause for example:
```rust
            Expr::BinaryOp {
                left,
                op: BinaryOperator::Eq,
                right,
            } if self.dialect.supports_eq_alias_assigment() && matches!(left.as_ref(), Expr::Identifier(_)) => {
                let Expr::Identifier(alias) = *left else {
                    return parser_err!(
                        "BUG: expected identifier expression as alias",
                        self.peek_token().location
                    );
                };
                Ok(SelectItem::ExprWithAlias {
                    expr: *right,
                    alias,
                })
            }
```
I think the least desirable option involves the expression cloning, i.e. the `right.clone()` I think we want to avoid in any case